### PR TITLE
849 display region or collection info drawer

### LINKED ISSUE #849: Regions: Display complete Region or Collection info in a drawer
When a user clicks a row in the Regions or Collections table, they should see the full details of the region in a drawer (see Verify, Captures, Growers, Earnings for inspiration).
Screenshot 2022-09-28 at 21 23 16
## Regions
* The region shape on a basic map
* Name
* Owner
* Collection (link to collection details)
* Properties
* Show on Org Map value
* Statistics Calculated value
* Download Shapefile button
* Edit button
* Delete button
## Collections
* All regions on a basic map
* Name
* Owner
* List of regions (each a link to region details)
* Edit button
* Delete button

### PR DESCRIPTION
**Pull request by marinakosova**
## Description

_I added Drawer for Regions section. It displays Region and Collection info._

https://user-images.githubusercontent.com/50210952/196255053-fd82ee59-0402-4e81-8ca3-6c58a77d5dbd.mov

**Issue(s) addressed**
- Resolves #849 

**What kind of change(s) does this PR introduce?**

- [x] Enhancement
- [ ] Bug fix
- [ ] Refactor

**Please check if the PR fulfils these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**
No Drawer for Regions or Collection table.

**What is the new behavior?**
When you click on any Regions or Collection table row, you'll see the full details of the region in a drawer.

## Breaking change

**Does this PR introduce a breaking change?**

No

## Other useful information

I recommend creating a new issue for adding [GeoJSON API](https://react-leaflet.js.org/docs/api-components/#geojson) data to Regions Drawer, since I didn't implemented it. For now, everything else works well, and user can experience this feature.

### CODE REVIEW

File: src/components/Regions.js [hunk 1]
```diff
@@ -304,6 +354,174 @@ const RegionTable = (props) => {
...
+                      onClick={(e) => {
+                        e.stopPropagation();
+                        setShowCollections(true);
+                        let region = selectedItem;
+                        let collection = region.collection;
+                        setSelectedItem(collection);
+                      }}
+                      style={{ display: 'inline-block', cursor: 'pointer' }}
+                    >
+                      Collection details
```
Comment (nmcharlton) [inline src/components/Regions.js:427]: Is it possible to display the name of the collection here?

File: src/components/Regions.js [hunk 2]
```diff
@@ -304,6 +354,174 @@ const RegionTable = (props) => {
...
+            )}
+
+            {!showCollections && (
+              <Grid container direction="column" justifyContent="space-around">
+                <Grid
+                  container
+                  direction="row"
+                  className={classes.itemRegionsDetail}
+                >
+                  <Typography>Show on Org Map?</Typography>
```
Comment (nmcharlton) [inline src/components/Regions.js:453]: Remove the question mark

**Review by nmcharlton — CHANGES_REQUESTED** [review]
This is great stuff. Thanks @marinakosova!
Just a couple of minor suggestions.

File: src/components/Regions.js (hunk 1 above)
Comment (marinakosova) [reply src/components/Regions.js:427]: Yes, @nmcharlton. Fixed it!

File: src/components/Regions.js (hunk 2 above)
Comment (marinakosova) [reply src/components/Regions.js:453]: @nmcharlton fixed it!

File: src/components/Regions.js (hunk 1 above)
Comment (marinakosova) [reply src/components/Regions.js:427]: Looks like this right now
Screen Shot 2022-10-26 at 2 37 26 PM

File: src/context/RegionContext.js [hunk 3]
```diff
@@ -95,7 +95,7 @@ export function RegionProvider(props) {
     });
     setRegions(regions);
     setRegionCount(count);
-
+    console.log(regions);
```
Comment (nmcharlton) [inline src/context/RegionContext.js:98]: If we need to leave logging in, use the `loglevel` utility (search the code base for existing uses).

**Review by nmcharlton — CHANGES_REQUESTED** [review]

File: src/context/RegionContext.js (hunk 3 above)
Comment (marinakosova) [reply src/context/RegionContext.js]: @nmcharlton sorry, missed this console log. Fixed it!

**Review by nmcharlton — APPROVED** [review]

**Pull request merged**